src/main/kotlin/com/strumenta/kolasu/model/Processing.kt

```diff
@@ -272,12 +272,45 @@ fun Node.mapTree(operation: (Node) -> Node): Node {
 }
 
 /**
- * Replace this node with "other" (by modifying the children of the parent node.)
- * For this to work, assignParents() must have been called.
+ * Replace [this] node with [other] (by modifying the children of the parent node.)
+ * For this to work, [Node.assignParents] must have been called.
  */
 fun Node.replaceWith(other: Node) {
     if (this.parent == null) {
         throw IllegalStateException("Parent not set")
     }
     this.parent!!.transformTree { if (it == this) other else it }
 }
+
+/**
+ * Looks for [oldNode] in the lists of nodes in this node.
+ * When found, it is removed, and in its place the [newNodes] are inserted.
+ */
+@Suppress("UNCHECKED_CAST") // assumption: a MutableList with a Node in it is a MutableList<Node>
```

Comment:
What happens if the property allows for one element and not a list?

Reply:
Nothing. Is that a valid use case or shall we make it an error?

Reply:
I would expect it to throw an error, it would be more clear to me